auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LdapRolesProvider.java

```diff
@@ -190,4 +191,25 @@ private void initialize() throws Exception {
         return roles;
     }
 
+    @Override
+    public LdapAuthHealth checkHealth() {
+
+        boolean isHealthy = true;
+        String details = null;
+        try {
+            LDAPConnection ldapConnection = connectionPool.getConnection();
```

Comment:
I checked the java doc for the API method in LDAPConnectionPoolHealthCheck class. These methods are anyway invoked when we call connectionPool.getConnection() or when a connection is released. A successful connection instance from pool is only returned if associated health checks pass otherwise it throws LDAPException. 

Do you suggest to make an explicit check on these methods again in our health check?